Source/WebCore/loader/cache/CachedImage.cpp

```diff
@@ -314,6 +315,13 @@ FloatSize CachedImage::imageSizeForRenderer(const RenderElement* renderer, SizeT
     if (!image)
         return { };
 
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+    if (renderer && renderer->isMultiRepresentationHEIC()) {
+        auto metrics = renderer->style().fontCascade().primaryFont().metricsForMultiRepresentationHEIC();
+        return { metrics.ideogramWidth().value_or(0) , metrics.height() };
```

Comment:
I am not sure why the width can't be retrieved but the height can be. But returning zero for the width does not seem right. Maybe `font.avgCharWidth()` is more appropriate.

Reply:
Oh I see you are using `FontMetrics`. But the meaning of `ideogram` of the font is different from the meaning of `MultiRepresentationHEIC`. See my comment below.